### PR TITLE
Add support for converting LOM to Learnosity tags (WBL-21)

### PR DESCRIPTION
**Pull request by rakeshqdsa**

### CODE REVIEW

File: src/Services/ConvertToLearnosityService.php [hunk 1]
```diff
@@ -99,13 +99,14 @@ public function process()
     private function parseContentPackage()
     {
         $manifestFolders = $this->parseInputFolders();
+        
         $finalManifest = $this->getJobManifestTemplate();
-
+
```
Comment (ttton) [inline src/Services/ConvertToLearnosityService.php]: PSR-2:
* Remove trailing whitespace from newline

File: src/Services/ConvertToLearnosityService.php [hunk 2]
```diff
@@ -212,11 +213,11 @@ private function getItemResourcesByHrefFromDocument(\DOMDocument $manifestDoc)
     {
         $itemResources = [];
         $resources = $manifestDoc->getElementsByTagName('resource');
-
+
```
Comment (ttton) [inline src/Services/ConvertToLearnosityService.php]: PSR-2: see [#diff-6f0e68dc90c0979d95f5acaf95f64b4eR104](https://github.com/Learnosity/learnosity-qti/pull/36/files#diff-6f0e68dc90c0979d95f5acaf95f64b4eR104)

File: src/Services/ConvertToLearnosityService.php [hunk 3]
```diff
@@ -212,11 +213,11 @@ private function getItemResourcesByHrefFromDocument(\DOMDocument $manifestDoc)
     {
         $itemResources = [];
         $resources = $manifestDoc->getElementsByTagName('resource');
-
+        
         while (($resource = $resources->item(0)) != null) {
             $resourceHref = $resource->getAttribute('href');
             $resourceType = $resource->getAttribute('type');
-
+
```
Comment (ttton) [inline src/Services/ConvertToLearnosityService.php]: PSR-2: see [#diff-6f0e68dc90c0979d95f5acaf95f64b4eR104](https://github.com/Learnosity/learnosity-qti/pull/36/files#diff-6f0e68dc90c0979d95f5acaf95f64b4eR104)

File: src/Services/ConvertToLearnosityService.php [hunk 4]
```diff
@@ -276,11 +278,36 @@ private function metadataIdentifierExists(\DOMNode $resource)
...
+        $xpath = new \DOMXPath($resource->ownerDocument);
+        $xpath->registerNamespace('lom', 'http://ltsc.ieee.org/xsd/LOM');
+        $xpath->registerNamespace('qti', 'http://www.imsglobal.org/xsd/imscp_v1p1');
+
+        $searchResult = $xpath->query('.//lom:taxonPath', $resource);
+        $itemTagsArray = array();
+        foreach($searchResult as $search){
+            $tagName = $xpath->query('.//lom:source/lom:string', $search)->item(0)->textContent."\n";
+            $tagValues = $xpath->query('.//lom:taxon/lom:entry/lom:string', $search)->item(0)->textContent."\n";
+            if(!empty(trim($tagValues))){
```
Comment (ttton) [inline src/Services/ConvertToLearnosityService.php]: PSR-2: see [#diff-6f0e68dc90c0979d95f5acaf95f64b4eR301](https://github.com/Learnosity/learnosity-qti/pull/36/files#diff-6f0e68dc90c0979d95f5acaf95f64b4eR301)

File: src/Services/ConvertToLearnosityService.php [hunk 5]
```diff
@@ -276,11 +278,36 @@ private function metadataIdentifierExists(\DOMNode $resource)
...
+     */
+    private function getTaxonPathEntryForItemTags(\DOMNode $resource)
+    {
+        $xpath = new \DOMXPath($resource->ownerDocument);
+        $xpath->registerNamespace('lom', 'http://ltsc.ieee.org/xsd/LOM');
+        $xpath->registerNamespace('qti', 'http://www.imsglobal.org/xsd/imscp_v1p1');
+
+        $searchResult = $xpath->query('.//lom:taxonPath', $resource);
+        $itemTagsArray = array();
+        foreach($searchResult as $search){
```
[the remark this answers is not in the document]
Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

File: src/Services/ConvertToLearnosityService.php (hunk 1 above)
Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

File: src/Services/ConvertToLearnosityService.php (hunk 2 above)
Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

File: src/Services/ConvertToLearnosityService.php (hunk 3 above)
Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

File: src/Services/ConvertToLearnosityService.php (hunk 4 above)
Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

File: src/Services/ConvertToLearnosityService.php [hunk 6]
```diff
@@ -96,16 +96,15 @@ public function process()
      * Performs a conversion on each directory (one level deep)
      * inside the given source directory.
      */
-    private function parseContentPackage()
-    {
+    private function parseContentPackage() {
```
Comment (ttton) [inline src/Services/ConvertToLearnosityService.php]: PSR-2: The opening brace should be on a newline for method declarations (looks good elsewhere though):
https://www.php-fig.org/psr/psr-2/#43-methods

File: src/Services/ConvertToLearnosityService.php [hunk 7]
```diff
@@ -209,15 +208,14 @@ private function convertQtiContentPackagesInDirectory($sourceDirectory, $relativ
      *
      * @return array
      */
-    private function getItemResourcesByHrefFromDocument(\DOMDocument $manifestDoc)
-    {
+    private function getItemResourcesByHrefFromDocument(\DOMDocument $manifestDoc) {
```
Comment (ttton) [inline src/Services/ConvertToLearnosityService.php]: PSR-2: Same issue with brace as here: https://github.com/Learnosity/learnosity-qti/pull/36/files/3bca6cc3a4d9aec3eef72eed8b2c6924492bb268#diff-6f0e68dc90c0979d95f5acaf95f64b4eR99

File: src/Services/ConvertToLearnosityService.php [hunk 8]
```diff
@@ -290,18 +288,17 @@ private function metadataIdentifierExists(\DOMNode $resource)
      *
      * @return array
      */
-    private function getTaxonPathEntryForItemTags(\DOMNode $resource)
-    {
+    private function getTaxonPathEntryForItemTags(\DOMNode $resource) {
```
Comment (ttton) [inline src/Services/ConvertToLearnosityService.php]: PSR-2: Same issue as https://github.com/Learnosity/learnosity-qti/pull/36/files/3bca6cc3a4d9aec3eef72eed8b2c6924492bb268#diff-6f0e68dc90c0979d95f5acaf95f64b4eR99

**Comment by ttton** [comment]
Looks good, just a couple of last comments regarding PSR-2 on method declarations.

File: src/Services/ConvertToLearnosityService.php (hunk 6 above)
Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

File: src/Services/ConvertToLearnosityService.php (hunk 7 above)
Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

File: src/Services/ConvertToLearnosityService.php (hunk 8 above)
Comment (rakeshqdsa) [reply src/Services/ConvertToLearnosityService.php]: Done

**Comment by rakeshqdsa** [comment]
Done

**Pull request merged**